kernel/src/process/posix_thread/futex.rs

```diff
@@ -406,6 +374,32 @@ fn get_futex_bucket(key: &FutexKey) -> (usize, &'static SpinLock<FutexBucket>) {
     FUTEX_BUCKETS.get().unwrap().get_bucket(key)
 }
 
+/// Lock futex buckets in a consistent order to avoid deadlock.
```

Comment:
Since the code is extracted to a function, I think a richer docstring would be really helpful
```suggestion
/// Locks the futex buckets associated with a given pair of keys in a consistent
/// order to avoid deadlock.
///
/// The order is defined by comparing the respective bucket indices and then locking
/// the one with the lowest index first. If both keys belong to the same bucket,
/// lock it and return the second lock guard as `None`.
```